ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+	d.Set("device_interface_id", tenantConnection.DeviceInterfaceID)
+	d.Set("port_id", tenantConnection.PortID)
+	d.Set("status", tenantConnection.Status)
+
+	return nil
+}
+
+func resourceProviderConnectivityTenantConnectionV2Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	var hasChange bool
+	var updateOpts tenant_connections.UpdateOpts
+
+	if d.HasChange("name") {
+		hasChange = true
+		updateOpts.Name = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		hasChange = true
+		updateOpts.Description = d.Get("description").(string)
+	}
+
+	if d.HasChange("tags") {
+		hasChange = true
+		updateOpts.Tags = getTags(d, "tags")
+	}
+
+	if d.HasChange("name_other") {
+		hasChange = true
+		updateOpts.NameOther = d.Get("name_other").(string)
+	}
+
+	if d.HasChange("description_other") {
+		hasChange = true
+		updateOpts.DescriptionOther = d.Get("description_other").(string)
+	}
+
+	if d.HasChange("tags_other") {
+		hasChange = true
+		updateOpts.TagsOther = getTags(d, "tags_other")
+	}
+
+	if hasChange {
+		r := tenant_connections.Update(client, d.Id(), updateOpts)
+		if r.Err != nil {
+			return fmt.Errorf("error updating ECL Provider Connectivity Tenant Connection: %s", r.Err)
+		}
+		log.Printf("[DEBUG] Tenant Connection has successfully updated.")
+	}
+
+	return resourceProviderConnectivityTenantConnectionV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	tenantConnection, err := tenant_connections.Get(client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "Provider Connectivity Tenant Connection")
+	}
+
+	deviceType := tenantConnection.DeviceType
+
+	if err := tenant_connections.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("error deleting ECL Provider Connectivity Tenant Connection: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{"DELETED"},
+		Refresh:    waitForTenantConnectionStateDelete(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	if deviceType == "ECL::VirtualNetworkAppliance::VSRX" {
+		stateConf.PollInterval = vnaUpdatePollInterval
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting ECL Tenant Connection: %s", err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func getServerAttachmentOpts(d *schema.ResourceData) (tenant_connections.Server, error) {
```

Comment:
This function does not return any error actually. Please consider again if any error will happen or not by invalid parameter and so on. If never happen, you can remove error from return values.

Reply:
Yes, confirm.